dxr/plugins/clang/menus.py

```diff
@@ -118,31 +118,40 @@ def _more_menu_items(self, (macro_name,)):
 class TypeRef(_QualnameRef):
     @classmethod
     def _condensed_menu_data(cls, tree, prop):
-        """Hang onto qualname *and* kind."""
         return (super(TypeRef, cls)._condensed_menu_data(tree, prop) +
-                (prop.get('kind', ''),))
+                (prop.get('kind', ''), 'has_subclass' in prop, 'has_base_class' in prop))
 
-    def _more_menu_items(self, (qualname, kind)):
+    def _more_menu_items(self, (qualname, kind, has_subclass, has_base_class)):
         """Return menu for type reference."""
+        def kind_plural():
+            if kind == 'class':
+                return 'classes'
+            elif kind == 'struct':
+                return 'structs'
+            raise TypeError('Unexpected kind: %s' % kind)
```

Comment:
I'd rather move this to indexing time, as people using the web interface should never see exceptions. Also, does it need to be a fatal exception? It seems like we could fall back to a more generic description like "type" or "thing".